pkg/api/converterfromplugin.go

```diff
@@ -0,0 +1,65 @@
+package api
+
+import (
+	plugin "github.com/openshift/openshift-azure/pkg/api/plugin/api"
+)
+
+// ConvertFromPlugin converts external plugin API config type into
+// internal API Config type
+func ConvertFromPlugin(in *plugin.Config, old *Config) Config {
+	cs := &Config{}
+	// preserve old config if this is update
+	if old != nil {
```

Comment:
old is never going to be nil the way you use it now (see my earlier comment) but this is probably fine since `cs` is either old or a pointer to an empty struct?

Reply:
hmm. Should we just skip `old !=nil` and `deepCopy` always?

Reply:
I like the nil check protection here.